src/Microsoft.AspNet.OData.Shared/Common/ODataPathHelper.cs

```diff
@@ -0,0 +1,56 @@
+//-----------------------------------------------------------------------------
+// <copyright file="ODataPathHelper.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData.Common
+{
+    /// <summary>
+    /// Helper methods for <see cref="ODataPath"/>.
+    /// </summary>
+    public static class ODataPathHelper
```

Comment:
Is this meant to be public?

Reply:
Yes - for use by the customers using the nuget package